client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/Forms/ExposureForm/ExposureForm.tsx

```diff
@@ -0,0 +1,225 @@
+import { useSelector } from 'react-redux';
+import React, { useState, useEffect } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { CircularProgress, Grid, MenuItem } from '@material-ui/core';
+
+import Map from 'commons/Map/Map';
+import useFormStyles from 'styles/formStyles';
+import PlaceSubType from 'models/PlaceSubType';
+import CovidPatient from 'models/CovidPatient';
+import DatePick from 'commons/DatePick/DatePick';
+import StoreStateType from 'redux/storeStateType';
+import ExposureFields from 'models/enums/ExposureFields';
+import FormRowWithInput from 'commons/FormRowWithInput/FormRowWithInput';
+import ExposureSearchTextField from 'commons/AlphabetTextField/ExposureSearchTextField';
+import PlacesTypesAndSubTypes from 'commons/Forms/PlacesTypesAndSubTypes/PlacesTypesAndSubTypes';
+
+import useStyles from './ExposureFormStyles';
+import useExposureForm from './useExposureForm';
+import ExposureSourceOption from './ExposureSourceOption';
+import { FormData } from '../../ExposuresAndFlightsInterfaces';
+import { Exposure } from 'commons/Contexts/ExposuresAndFlights';
+
+const INSERT_EXPOSURE_SOURCE_SEARCH = 'הזן שם פרטי, שם משפחה, מספר זיהוי או מספר טלפון';
+const MAX_DATE_ERROR_MESSAGE = 'לא ניתן להזין תאריך מאוחר מתאריך תחילת המחלה';
+const INVALID_DATE_ERROR_MESSAGE = 'תאריך לא חוקי';
+
+export const phoneAndIdentityNumberRegex = /^([\da-zA-Z]+)$/;
+
+interface Props {
+	exposureAndFlightsData: any; 
+	fieldsNames: any; 
+	handleChangeExposureDataAndFlightsField: (
+        index: number,
+        fieldName: string,
+        value: any
+    ) => void;
+	index : number; 
+}
+
+const ExposureForm = (props: Props) => {
+	const { exposureAndFlightsData, fieldsNames, handleChangeExposureDataAndFlightsField, index } = props;
+
+	const classes = useStyles();
+	const formClasses = useFormStyles();
+
+	const { control , setValue , errors } = useFormContext();
+
+	const validationDate = useSelector<StoreStateType, Date>((state) => state.investigation.validationDate);
+
+	const [exposureSourceSearchString, setExposureSourceSearchString] = useState<string>('');
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [optionalCovidPatients, setOptionalCovidPatients] = useState<CovidPatient[]>([]);
+
+	const epidemiologyNumber = useSelector<StoreStateType, number>((state) => state.investigation.epidemiologyNumber);
+
+	const { fetchOptionalCovidPatients, selectedExposureSourceDisplay } = useExposureForm({
+		exposureAndFlightsData,
+		exposureSourceSearchString,
+	});
+	useEffect(() => {
+		const setOptionalCovidPatientsAsync = async () => {
```

Comment:
it's more apparent in #1025 where I've separated its scope to outside of useEffect but this function has to be async because I want it to get the optionalcovidpatients from fetchOptionalCovidPatients (with await) and set it ( I've seperated new function to maintain single responsibility (one function for getting and one for setting)